tchannel/peer_strategy.py

```diff
@@ -30,3 +32,23 @@ def get_score(self, peer):
 class ZeroScoreCalculator(ScoreCalculator):
     def get_score(self, peer):
         return 0
+
+
+class PreferIncomingCalculator(ScoreCalculator):
+    def get_score(self, peer):
+        """Calculate the peer score based on connections.
+
+        If the peer has no incoming connections, it will have largest score.
+        In our peer selection strategy, the largest number has least priority
+        in the heap.
+
+        If the peer has incoming connections, we will return number of outbound
+        pending requests and responses.
+
+        :param peer: instance of `tchannel.tornado.peer.Peer`
+        :return: score of the peer
+        """
+        if not peer.incoming_connections:
```

Comment:
Do we need to make sure that the peer is not 0.0.0.0:0. I don't think we're checking that right now and that could cause problems with tcurl.